src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -130,6 +131,16 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
                     var descriptor = new ServiceDescriptor(serviceName) { HostGroup = hostGroup };
                     return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
                 }
+                catch (OperationCanceledException)
+                {
+                    // if it is our own cancellation token, then rethrow
+                    // otherwise, let us retry.
```

Comment:
> I think if timeout is part of contract, it should accept timeout value or behavior flag and/or should throw timeout exception rather than cancellation exception.

That's a very fair point. I'd add "Or document it" to the list of options. Please file a bug against us to consider this.